### DISCUSSION TITLE
Support for torrent injection

### DISCUSSION THREAD

**Discussion by mat926** [RogueOneEcho/caesura#90] [Ideas]
Instead of saving to torrent files, can we have the torrents injected into the client directly?

**Comment by RogueOneEcho** [comment]
This is something I'd like but it's a pain to support each torrent client, the configurations required for them, and the quirks/errors of each.

I've worked with the Deluge API a number of times and it's an absolute pain because it'll seemingly randomly disconnect and leads to a lot of confusion.

As an example this is the logic I use to set a listen port in Deluge. It should be a single command but ended up being a whole series of them to cover different edge cases: https://github.com/RogueOneEcho/how-to-setup-deluge-with-protonvpn-portforward/blob/ec9459296c44a322fd4ca076e0c7d96feab1c927/monitor/monitor#L76-L162

If I were to add direct injection it would likely only be for:
- [ ] Deluge
- [ ] qBittorrent

**Reply by RogueOneEcho** [reply]
related: #91

**Reply by RogueOneEcho** [reply]
- [ ] Investigate fast resume

I seem to recall reading that either cross-seed or fertilizer were doing some magic to set the fastresume file as well that meant the torrent recheck could be skipped.

**Reply by RogueOneEcho** [reply]
Update:

A long while ago I developed https://github.com/RogueOneEcho/tremolo 
In order to safely move all my torrents from deluge to qbit.

For that I also built some basic APIs for qBit and deluge:

- https://github.com/RogueOneEcho/qbittorrent_api
- https://github.com/RogueOneEcho/deluge_api

I had intended to also use those clients to implement #90 but never got around to it.

My preference for torrent injection would use those clients, though they may need a fair bit of work to update them to the latest patterns.

**Reply by tomerh2001** [reply]
@RogueOneEcho I can try take on the qBittorrent part since I need it, I'll see what I can do

**Reply by RogueOneEcho** [reply]
Completed in #206
